File: Test/OptionalTests.m
//
//  MuscleAssertOptionalTests.swift
//  MuscleAssert
//
//  Created by akuraru on 2017/01/02.
//
//

#import <XCTest/XCTest.h>
#import "MuscleAssert/MuscleAssert.h"

@interface OptionalTests : XCTestCase
@property (nonatomic) MuscleAssert *assert;
@end

@implementation OptionalTests

- (void)setUp {
    [super setUp];
    self.assert = [[MuscleAssert alloc] init];
}

- (void)testNone {
    NSString *diff = [self.assert deepStricEqual:nil right:nil];
    XCTAssertNil(diff);
}

- (void)testLeftNone {
    NSString *diff = [self.assert deepStricEqual:nil right:@"abc"];
    XCTAssertEqualObjects(diff, @"\npath: .Optional\n"
                          "  left: value is none\n"
                          "  right: abc\n");
}

- (void)testEmptyRight {
    NSString *diff = [self.assert deepStricEqual:@"abc" right:nil];
    XCTAssertEqualObjects(diff, @"\npath: .Optional\n"
                          "  left: abc\n"
                          "  right: value is none\n");
}

@end

File: SwiftTest/OptionalTests.swift
//
//  OptionalTests.swift
//  SwiftTest
//
//  Created by akuraru on 2017/10/21.
//

import XCTest
@testable import MuscleAssert

class OptionalTests: XCTestCase {
    let assert = MuscleAssert()
    
    func testNone() {
        let diff = assert.deepStricEqual(left: nil, right: nil)
        XCTAssertNil(diff);
    }
    
    func testLeftNone() {
        let diff = assert.deepStricEqual(left: nil, right: "abc")
        XCTAssertEqual(diff, "\npath: .Optional\n" +
            "  left: value is none\n" +
            "  right: abc\n")
    }
    
    func testEmptyRight() {
        let diff = assert.deepStricEqual(left: "abc", right: nil)
        XCTAssertEqual(diff, "\npath: .Optional\n" +
            "  left: abc\n" +
            "  right: value is none\n")
    }
}
